launcher/ui/themes/ThemeManager.cpp

```diff
@@ -28,14 +28,16 @@
 
 ThemeManager::ThemeManager(MainWindow* mainWindow) {
     m_mainWindow = mainWindow;
+    InitializeThemes();
 }
 
 /// @brief Adds the Theme to the list of themes
 /// @param theme The Theme to add
 /// @return Theme ID
-QString ThemeManager::AddTheme(ITheme * theme) {
-    m_themes.insert(std::make_pair(theme->id(), std::unique_ptr<ITheme>(theme)));
-    return theme->id();
+QString const& ThemeManager::AddTheme(std::unique_ptr<ITheme> theme) {
+    QString const& id = theme->id();
+    m_themes.emplace(id, std::move(theme));
+    return id;

```

Comment:
Oh wait, `theme->id()` returns a copy :/
didn't know that, so you can't return id by ref. either make `id()` return a ref or return by copy, your crash is probably here. sorry for the bad suggestion

Reply:
all good, its fixed now hopefully at least. ^^